K8s-deployment/Charts/databroker/README.md

```diff
@@ -0,0 +1,442 @@
+# RabbitMQ cluster in k8s as a StatefulSet
+
+## Create secret files
+
+Make a copy of sample secrets directory and add appropriate values to all files.
+
+```console
+ cp -r example-secrets/secrets .
+```
+
+```
+# secrets directory after generation of secret files
+```sh
+secrets/
+├── credentials
+|   ├── .erlang.cookie (Random characters - 50)
+│   ├── rabbitmq-admin-passwd
+│   └── rabbitmq-definitions.json
+└── pki
+    ├── rabbitmq-ca-cert.pem (letsencrpt chain.pem)
+    ├── rabbitmq-server-cert.pem (letsencrpt fullchain.pem)
+    └── rabbitmq-server-key.pem (letsencrpt privkey.pem)
+
+```
+
+## Define Appropriate values of resources
+
+Define Appropriate values of resources -
+- CPU request and limits
+- RAM request and limits
+
+in `resource-values.yaml` as shown in sample resource-values file for [`aws`](./example-aws-resource-values.yaml) and [`azure`](./example-azure-resource-values.yaml)
+
+## Required storage class
+```sh
+# For ebs as storage class, deploy ebs-storage class in K8s cluster if not present
+kubectl apply -f ../../K8s-cluster/addons/storage/aws/ebs-storageclass.yaml
+```
+## Required cluster-autoscaler
+```sh
+# Deploy cluster-autoscaler if not present in K8s cluster
+kubectl apply -f ../../K8s-cluster/cloud-specific/aws/cluster-autoscaler-autodiscover.yaml
+```
+## Deploy
+## Installing the Chart
+
+To install the `rabbitmq` chart:
+
+```console
+./install.sh
+```
+The command deploys rabbitmq on the Kubernetes cluster in a 3 node configuration. The Parameters section lists the parameters that can be configured during installation.
+
+
+The script will create :
+1. create a namespace `rabbitmq`
+2. create required secrets
+3. deploy rabbitmq statefulset and services
+
+#### Helpers
+```sh
+# Pod status, IP, node, etc
+kubectl -n rabbitmq get pods -o wide
+
+# RabbitMQ pod logs
+kubectl logs -n rabbitmq -f POD_NAME
+```
+## Uninstalling the Chart
+
+To uninstall/delete the `rabbitmq` deployment:
+
+```console
+helm uninstall rabbitmq -n rabbitmq
+```
+The command removes all the Kubernetes components associated with the chart and deletes the release.
+
+
+To delete rabbitmq 'pvc'
+```console
+kubectl delete pvc data-rabbitmq-X -n rabbitmq
+```
+## To dos:
+1. Integration of cert-manager and TLS certs of RMQ
+2. Prometheus operator integration with RMQ monitoring
+3. Differentiate following things between environements - dev(mostly deployed in minikube), test and production mostly using Kustomize:
+   - Persistent Volumes (PV)  
+   - resource requests and limits
+   - namespace naming
+## References
+1. [RabbitMQ in k8s using StatefulSet in Minikube](https://github.com/rabbitmq/diy-kubernetes-examples/tree/master/minikube)
+2. [Rabbitmq in k8s using Statefulset in cloud-specific](https://github.com/rabbitmq/diy-kubernetes-examples/tree/master/gke)
+3. [Cluster Autoscaler](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler/cloudprovider/aws)
+
+## Parameters
```

Comment:
You can just point to bitnami helm chart . No need to have params here

Reply:
Removed.